.github/workflows/rawls-run-azure-e2e-tests.yaml

```diff
@@ -275,7 +286,7 @@ jobs:
       - init-github-context
       - rawls-swat-e2e-test-job
       - delete-billing-project-v2-from-bee-workflow
-    if: ${{ needs.init-github-context.outputs.delete-bee && always() }} # always run to confirm bee is destroyed unless explicitly requested not to
+    if: ${{ needs.init-github-context.outputs.delete-bee == 'true' }}
```

Comment:
Examples of the `delete-bee` option working:

<img width="520" alt="image" src="https://github.com/broadinstitute/rawls/assets/484484/95f2bbda-609a-4639-9f22-8c2eb5ef2a63">
<img width="514" alt="image" src="https://github.com/broadinstitute/rawls/assets/484484/5709e38a-09e4-4fa7-bd18-c60b92f6aa7f">

vs.
<img width="437" alt="image" src="https://github.com/broadinstitute/rawls/assets/484484/05a6ca21-6e15-4e6a-904e-e81dd1db8d1a">
<img width="368" alt="image" src="https://github.com/broadinstitute/rawls/assets/484484/422fbf14-8a9a-455b-b792-23421fa0a410">